MonoGame.Framework/Graphics/Texture.cs

```diff
@@ -4,14 +4,18 @@
 
 using System;
 using System.Diagnostics;
+using System.Threading;
 
 namespace Microsoft.Xna.Framework.Graphics
 {
-	public abstract partial class Texture : GraphicsResource
+	public abstract partial class Texture : GraphicsResource, IComparable<Texture>
 	{
 		internal SurfaceFormat _format;
 		internal int _levelCount;
 
+        private readonly int _comparisonCode = Interlocked.Increment(ref _lastComparisonCode);
+        private static int _lastComparisonCode;

```

Comment:
Was thinking maybe we want this to default to `Int32.MinValue`.  This way we don't get a sudden change in sort order if the count ever rolls over from `Int32.MaxValue`.

I was curious and it would take creating one new texture per second for over 136 years for this value to fully roll over.  So i'm not very concerned about the numerical limits here.